### PR TITLE
feat: add prepare to work with monorepo script

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build 0fcc361 — [Extension build](https://github.com/leather-io/extension/actions/runs/10560054635), [Test report](https://leather-io.github.io/playwright-reports/feat-mono-prepare-script), [Storybook](https://feat-mono-prepare-script--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=feat-mono-prepare-script)

This pr adds `mono:prepare` command and script to make extension ready to work with monorepo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,8 @@
     "storybook-docs": "storybook dev --docs",
     "build-storybook": "storybook build",
     "chromatic": "npx chromatic --project-token=CHROMATIC_PROJECT_TOKEN",
-    "preinstall": "npx only-allow pnpm"
+    "preinstall": "npx only-allow pnpm",
+    "mono:prepare": "rm -rf ./node_modules && node scripts/transform-mono.js && pnpm install"
```
Comment (kyranjamie) [inline package.json]: Is it necessary to clear node modules every time?

**Comment by pete-watters** [comment]
Thanks for doing this @alter-eggo 👍 I will help to test it. 

Based on the snippet [here](https://trustmachines.slack.com/archives/C05LAP952E7/p1724165168853399?thread_ts=1724164960.485939&cid=C05LAP952E7) it seems like this is adding everything to both dependancies and overrides which I'm not sure will work. 

When working on a package, the active package you are working on should be in dependancies and then the packages it depends on should be in overrides. [See here](https://github.com/leather-io/mono/blob/dev/docs/extension-development.md)

**Comment by pete-watters** [comment]
@fbwoolf @edgarkhanzadian : can you please test this when you have some time? It's working for @alter-eggo but failing for me and I cannot build `mono/packages/stacks` for some reason. 

It would be good if everyone can check this and see if it helps when working on mono repo packages in the extension

**Comment by pete-watters** [comment]
I was re-testing this now once I had solved my issues with the stacks package. I was unable to see changes from the UI package when working on it. 

I will test it more tomorrow and report back with more information

File: scripts/mono-link-ui.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      packageJson.pnpm.overrides = {};
+    }
+
+    // Add the specified overrides
+    const overrides = {
+      '@leather.io/rpc': 'file:../mono/packages/rpc',
+      '@leather.io/constants': 'file:../mono/packages/constants',
+      '@leather.io/models': 'file:../mono/packages/models',
+      '@leather.io/tokens': 'file:../mono/packages/tokens',
+      '@leather.io/utils': 'file:../mono/packages/utils'
```
Comment (fbwoolf) [inline scripts/mono-link-ui.js:38]: Are these all the overrides needed? I typ override them all, but maybe that is overkill?

**Review by fbwoolf — APPROVED** [review]

File: scripts/mono-link-ui.js (hunk 2 above)
Comment (alter-eggo) [reply scripts/mono-link-ui.js:38]: not sure here tbh, need to test

**Pull request merged**